Microsoft.Toolkit.Uwp.UI.Controls/AdaptiveGridView/AdaptiveGridView.cs

```diff
@@ -176,25 +181,37 @@ private void DetermineOneRowMode()
                         itemsWrapGridPanel.Orientation = Orientation.Vertical;
                     }
 
-                    this.SetBinding(GridView.MaxHeightProperty, b);
+                    SetBinding(MaxHeightProperty, b);
+
+                    _savedHorizontalScrollMode = ScrollViewer.GetHorizontalScrollMode(this);
+                    _savedVerticalScrollMode = ScrollViewer.GetVerticalScrollMode(this);
+                    _savedHorizontalScrollBarVisibility = ScrollViewer.GetHorizontalScrollBarVisibility(this);
+                    _savedVerticalScrollBarVisibility = ScrollViewer.GetVerticalScrollBarVisibility(this);
+                    _needToRestoreScrollStates = true;
 
                     ScrollViewer.SetVerticalScrollMode(this, ScrollMode.Disabled);
-                    ScrollViewer.SetVerticalScrollBarVisibility(this, ScrollBarVisibility.Disabled);
+                    ScrollViewer.SetVerticalScrollBarVisibility(this, ScrollBarVisibility.Hidden);
                     ScrollViewer.SetHorizontalScrollBarVisibility(this, ScrollBarVisibility.Visible);
                     ScrollViewer.SetHorizontalScrollMode(this, ScrollMode.Enabled);
                 }
                 else
                 {
-                    this.ClearValue(GridView.MaxHeightProperty);
+                    ClearValue(MaxHeightProperty);
                     if (itemsWrapGridPanel != null)
                     {
                         itemsWrapGridPanel.Orientation = Orientation.Horizontal;
                     }
 
-                    ScrollViewer.SetVerticalScrollMode(this, ScrollMode.Enabled);
-                    ScrollViewer.SetVerticalScrollBarVisibility(this, ScrollBarVisibility.Visible);
-                    ScrollViewer.SetHorizontalScrollBarVisibility(this, ScrollBarVisibility.Disabled);
-                    ScrollViewer.SetHorizontalScrollMode(this, ScrollMode.Disabled);
+                    if (!_needToRestoreScrollStates)
+                    {
+                        return;
+                    }
+
+                    _needToRestoreScrollStates = false;
```

Comment:
What happens when the changes the style to scroll horizontally instead of vertically?